nexus/tests/integration_tests/instances.rs

```diff
@@ -2921,6 +2925,9 @@ async fn test_instance_serial(cptestctx: &ControlPlaneTestContext) {
             .await;
 
     // Now, simulate completion of instance boot and check the state reported.
+    // TESTABILITY HACK: since we told sled-agent-sim to run a mock propolis,
+    // it will also tell the nexus that its propolis IP is localhost as part
+    // of this exchange. (nexus's call to sled-agent instance_put)
```

Comment:
I think I'll rephrase it to make it more clear, but basically: prior to this step, nexus's database is "wrong" about propolis's IP, because nexus "assigned" it one with its usual logic.
specifically how in the call to instance_simulate -> instance_finish_transition:
  - nexus makes API call to sled-agent-sim's instance_poke to tell it to pretend the instance finished booting
  - sled-agent-sim makes API call to nexus's cpapi_instances_put with the updated InstanceRuntimeState that *now also contains the mock propolis server's IP* which overwrites the one nexus had allocated via its usual instance-creating saga
  - *now* we can successfully connect to nexus's serial console endpoints.
 
This felt unusual enough to me that I wanted to write something down here to call it out, as `instance_simulate` otherwise would strike me as only being used for changing the runtime state of the instance, and not hiding any additional load-bearing magic like this -- and of course this is specific to the test use case where Nexus *would* be immediately wrong about the IP it allocated for the instance in the first place.